www/source/blog/2018-07-25-Packaging-COTS-for-Windows.html.md

```diff
@@ -0,0 +1,398 @@
+1. Packaging commercial off the shelf (COTS) applications usually purchased from a third party vendor.
+
+Habitat plan development patterns and issues to be considered at build and run time vary greatly between these two types of packaging. When it comes to the second class of packages - COTS - the techniques used in Linux are very different from Windows. Linux COTS packaging requires some understanding of ELF headers and patching them to point to Habitat dependencies. Windows binaries do not have ELF headers and use a different means of dynamic linking but there are many subtle and not so subtle nuances involved in packaging COTS applications very specific to Windows.
+
+This post will discuss different high level approaches to take with Windows COTS applications and dive into some details on how to create a Habitat package that is isolatable, portable, and results in the fewest amount of system wide side effects as possible when running in a Habitat Supervisor.
+
+## Application Compatibility
+
+There are two prerequisites that must be met to ensure that a COTS application can run in a Habitat Supervisor.
+
+### The Application Must Have a Silent Installer
+
+This is almost never a problem. The vast majority of applications are capable of being silently installed. If it does not have a silent installer, you may be able to get by if you choose an approach of packaging without the installer. Read on for more about that.
+
+If you must install via the application installer program and it has no silent install option, there is no automation tool, Habitat included, that can help.
+
+### The Application Must be Capable of Running on the Target OS
+
+Again, this is usually not a problem. Windows does a great job when it comes to backwards compatibility and many Windows Server 2003 applications will run on Windows server 2016. However there are absolutely legacy applications that cannot run on modern versions of Windows. Habitat does not employ any extra virtualization/emulation layer to make an application think it is running on an older operating system. For instance, 16 bit applications cannot run on a modern Windows version. There are other applications that may have hard coded checks to ensure they are running on a specific version or within a specific range of versions of Windows. SQL Server 2000 is an example of such an application. While it does not happen often, some Windows kernel APIs can change or be removed over time. If an application calls a Win32 API function with an outdated signature, that application may crash on a modern operating system.
+
+## Purity vs. Pragmatism
+
+When developing a Windows Habitat package for a COTS application, there are typically one of two ways to approach the `plan.ps1` and hooks:
+
+1. **A Purist approach**: Package run time artifacts remain in the Habitat file tree and do not affect external Habitat system state. Developing these can be challenging but have lots of benefits.
```

Comment:
I like both of these, but they differ (#1 reads in the third person, #2 in the first), so maybe pick one for both? (I might give a slight preference to third-person, just 'cause more standard, although first is kinda fun -- maybe put it in quotes? Or just disregard everything I'm saying? That'd be fine, too. 😄 )